sdks/java/extensions/euphoria/euphoria-core/src/main/java/org/apache/beam/sdk/extensions/euphoria/core/translate/BeamFlow.java

```diff
@@ -135,6 +137,69 @@ public static BeamFlow of(String name, Pipeline pipeline) {
     return ret;
   }
 
+  /**
+   * Registers the provided {@link Coder} for the given class.
+   *
+   * <p>
+   * Consider using
+   * {@link org.apache.beam.sdk.extensions.euphoria.core.translate.coder.RegisterCoders}
+   * when registering more coders at once.
+   * </p>
+   * @param clazz class of element to be encoded/decoded
+   * @param coder coder to encode/decode instances of given class
+   * @param <T> type parameter of the class to be encoded
+   */
+  public <T> void registerCoder(Class<T> clazz, Coder<T> coder) {
+    pipeline.getCoderRegistry().registerCoderForClass(clazz, coder);
+  }
+
+  /**
+   * Registers the provided {@link Coder} for the given type.
+   *
+   * <p>
+   * Consider using
+   * {@link org.apache.beam.sdk.extensions.euphoria.core.translate.coder.RegisterCoders}
+   * when registering more coders at once.
+   * </p>
+   * @param typeDescriptor type of element to be encoded/decoded
+   * @param coder coder to encode/decode instances of given class
+   * @param <T> type parameter of the class to be encoded
+   */
+  public <T> void registerCoder(TypeDescriptor<T> typeDescriptor, Coder<T> coder) {
+    pipeline.getCoderRegistry().registerCoderForType(typeDescriptor, coder);
+  }
+
+  /**
+   * Registers the provided {@link Coder} for the given class.
+   *
+   * <p>
+   * Consider using
+   * {@link org.apache.beam.sdk.extensions.euphoria.core.translate.coder.RegisterCoders}
+   * when registering more coders at once.
+   * </p>
+   *
+   * @param coderProvider
+   */
+  public void registerCoder(CoderProvider coderProvider) {
+    pipeline.getCoderRegistry().registerCoderProvider(coderProvider);
+  }
+
+  /**
+   * Registers new {@link ClassAwareKryoCoder} for the given raw class.
+   *
+   * <p>
+   * Consider using
+   * {@link org.apache.beam.sdk.extensions.euphoria.core.translate.coder.RegisterCoders}
+   * when registering more coders at once.
+   * </p>
+   *
+   * @param clazz class of element to be encoded/decoded
+   * @param <T> type parameter of the class to be encoded
+   */
+  public <T> void registerKryoCoder(Class<T> clazz){
+    registerCoder(clazz, new ClassAwareKryoCoder<>(clazz));
```

Comment:
shouldn't we have reusable kryo coder instance? (which can deserialize more than one class) I reckon having a separate instance for each class would not work properly for nested classes.

Reply:
Yes, we will have it when [BEAM-4705](https://issues.apache.org/jira/browse/BEAM-4705) will be done.